File: 3995-gcd-of-odd-and-even-sums/gcd-of-odd-and-even-sums.java
class Solution {
    public int gcdOfOddEvenSums(int n) {
        int odd = 0;
        int even  = 0;
        int ak = 1;
        int kb = 2;
        for(int i=0; i<n; i++){
            odd += ak;
            ak +=2;
        }
        for(int i=0; i<n; i++){
            even += kb;
            kb +=2;
        }
        return GCDOf(odd, even);
    }
    static int GCDOf(int a, int b){
        if(b == 0) return a ;
        return GCDOf(b, a % b);
    }
}

File: 3995-gcd-of-odd-and-even-sums/gcd-of-odd-and-even-sums.js
/**
 * @param {number} n
 * @return {number}
 */
var gcdOfOddEvenSums = function(n) {
    let odd = 0
    let even = 0
    let ak = 1
    let kb = 2
    for(let i = 0; i<n; i++){
        even += kb
        kb += 2
    }
    for(let i = 0; i<n; i++){
        odd += ak
        ak += 2
    }
    return GCDOf(odd, even)
};

function GCDOf(a, b){
    if(b === 0) return a
    return GCDOf(b, a % b)
}